keps/sig-network/3705-cloud-node-ips/README.md

```diff
@@ -656,36 +323,46 @@ This section must be completed when targeting beta to a release.
 
 ###### How can a rollout or rollback fail? Can it impact already running workloads?
 
-<!--
-Try to be as paranoid as possible - e.g., what if some components will restart
-mid-rollout?
-
-Be sure to consider highly-available clusters, where, for example,
-feature flags will be enabled on some API servers and not others during the
-rollout. Similarly, consider large clusters and how enablement/disablement
-will rollout across nodes.
--->
+Assuming no drastic bugs (eg, the cloud provider assigns Node X's IP
+to Node Y for no obvious reason), just restarting the cloud provider
+or kubelet with the new feature enabled-but-unused cannot fail.
+
+The cloud provider will not "re-taint" an existing working node if its
+node IP annotation becomes invalid. Thus, if the administrator
+accidentally rolls out a kubelet config that does something completely
+wrong (eg, specifying a new secondary node IP value which is not
+actually one of that node's IPs) then the only effect would be that
+the cloud provider will log "Failed to update node addresses for node"
+for that node.
+
+The most likely failure mode would be that in the process of
+reconfiguring nodes to use the new feature, the administrator
+reconfigures them _incorrecty_. (In particular, if nodes were
+previously using auto-detected primary node IPs, and the administrator
+needs to switch them to manually-specified dual-stack node IPs, they
+might end up manually specifying wrong (but valid) primary IPs.) In
+that case, the cloud provider would accept the new node IP value and
+update the node's addresses, possibly resulting in disruption.
+However, this would only happen as each kubelet was reconfigured and
+restarted, so as long as the administrator did not roll out the new
+configurations to every node simultaneously without any
+sanity-checking, they should only break a single node.
 
 ###### What specific metrics should inform a rollback?
 
-<!--
-What signals should users be paying attention to when the feature is young
-that might indicate a serious problem?
--->
+There are no relevant metrics; however, the feature will only affect
+nodes that have been reconfigured to use the new feature, so it should
+be obvious to the administrator if the feature is not working
+correctly.
 
 ###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
 
-<!--
-Describe manual testing that was done and the outcomes.
-Longer term, we may want to require automated upgrade/rollback tests, but we
-are missing a bunch of machinery and tooling and can't do that now.
--->
+No. The behavior during upgrade and rollback is not especially
+different from the ordinary runtime behavior.
```

Comment:
I agree with this statement, but still, given that IPs are propagated to cloud-provider via the annotation (so technically we have some persistent state related to this feature), I think we should test it.

Manual test is good enough here (and can be done after feature freeze and updated with the description of the test then - https://github.com/kubernetes/enhancements/pull/3658/files is a great example).

Does that make sense to you? If so, can you please switch to a TODO for running it and updating this question before promoting the feature to beta?